processor/signoztailsampler/factory_test.go

```diff
@@ -27,11 +13,16 @@ import (
 	"go.opentelemetry.io/collector/consumer/consumertest"
 )
 
-func TestCreateDefaultConfig(t *testing.T) {
-	cfg := createDefaultConfig()
-	assert.NotNil(t, cfg, "failed to create default config")
-	assert.NoError(t, componenttest.CheckConfigStruct(cfg))
-}
+// ToDo(amol):  commented this from otel-contrib/tail sampler
+// it results in stack overflow. this is maily because policy group
+// has self reference in form of subpolicies.
+// hence componenttest.CheckConfigStruct fails to validate config
+// rather keeps going in cycles to check the struct.
+// func TestCreateDefaultConfig(t *testing.T) {
+// 	cfg := createDefaultConfig()
+// 	assert.NotNil(t, cfg, "failed to create default config")
+// 	assert.NoError(t, componenttest.CheckConfigStruct(cfg))
+// }
```

Comment:
This shouldn't be TODO for later.

Reply:
Handled this now. Removed the self-ref from policy and this test goes through ok.